### ISSUE TITLE
Periodically cleanup drafts with no case id

### ISSUE THREAD

**Issue by schoork** [rubyforgood/casa#5442]
## What type(s) of user does this feature affect?
None

## Description
#5414 introduced the potential for case contact drafts to not be attached to a Casa Case (or cases). In this situation, if a user leaves step 1 without selecting at least one casa case, they will not be able to return to that draft. 

We need to introduce a way to clean these up periodically. They should be any case contacts that meet the following criteria:

1. `draft_case_ids` is empty
2. case contact was created more than 1 day go

As part of this work, 

- You may need to introduce a tool to run rake tasks or jobs periodically. One such tool is the [whenever gem](https://github.com/javan/whenever) (but you could use a different one). 
- If introduces a tool (like above) then incorporate with #5328 to make sure this runs on the 15th of every month

## Screenshots of current behavior, if any
You can paste images on the clipboard here

## How to access the QA site
_Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin
- all_casa_admin1@example.com view site as an all casa admin
  - go to `/all_casa_admins/sign_in`  

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in [slack](http://bit.ly/3Quxc1Q) #casa channel to ask questions quickly. And [discord](https://discord.gg/qJcw2RZH8Q) for office hours (currently Tuesday 5-7pm Pacific), stakeholder news, and upcoming new issues.

**Comment by phonghpham** [comment]
I can work on this.

**Comment by schoork** [comment]
Thanks @phonghpham. It's not quite ready to be worked, since it needs the PR it mentions to be merged first. When it's ready, will circle back to see if you still want to do it.

**Comment by schoork** [comment]
@phonghpham The blocking issue and PR has been merged. So this is ready if you want it.

**Comment by andreLumor** [comment]
Hello! One question about this issue.
From what I could see in the controller, what defines a draft is status being different from "active". Wouldn't this change also delete old case contacts that have a "started" status?

**Comment by schoork** [comment]
There was an after party task that ran on deployment that set their status as active.

**Comment by schoork** [comment]
@andreLumor lib/tasks/deployment/20231125151610_set_case_contacts_as_active.rake

**Comment by andreLumor** [comment]
Perfect then! I can start working on the issue if you wish.

**Comment by andreLumor** [comment]
Hey @schoork, had some time issues, but now I can finish this task.
One thing I was wondering is the usage of the whenever gem, since it (and other cron gems) doesn't work with heroku apps. (https://github.com/javan/whenever?tab=readme-ov-file#heroku).
Wouldn't it be better to keep using heroku scheduler, as in the linked birthday reminder task?

**Comment by schoork** [comment]
> Hey @schoork, had some time issues, but now I can finish this task. One thing I was wondering is the usage of the whenever gem, since it (and other cron gems) doesn't work with heroku apps. (https://github.com/javan/whenever?tab=readme-ov-file#heroku). Wouldn't it be better to keep using heroku scheduler, as in the linked birthday reminder task?

Yeah, didn't know that about Heroku and cron, but Heroku scheduler sounds good.

**Comment by elasticspoon** [comment]
@schoork Where is the 1 day coming from? Is there any situation where a draft without a case ID might hang around for longer than that and be otherwise accessible? Me and Linda talked a bit about this today and really want to avoid a situation were useful data is deleted.

**Comment by schoork** [comment]
> @schoork Where is the 1 day coming from?
> 
> Is there any situation where a draft without a case ID might hang around for longer than that and be otherwise accessible? Me and Linda talked a bit about this today and really want to avoid a situation were useful data is deleted.

I think it was something Brandon and I may have discussed (or maybe I just made it up). I'm totally cool with extending it, or even having a database flag where a user could mark it for deletion. Also, if this is not an issue for users, then we should just close it as not doing.

**Comment by Smullz622** [comment]
I would like to work on this

**Comment by schoork** [comment]
> I would like to work on this

It is yours, @Smullz622!

**Comment by Smullz622** [comment]
Still on hold until stakeholders confirm all drafts should be cleared after one week
